api/versionedsignedvalidatorregistration.go

```diff
@@ -0,0 +1,81 @@
+// Copyright © 2022 Attestant Limited.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"errors"
+
+	apiv1 "github.com/attestantio/go-eth2-client/api/v1"
+	"github.com/attestantio/go-eth2-client/spec"
+	"github.com/attestantio/go-eth2-client/spec/bellatrix"
+	"github.com/attestantio/go-eth2-client/spec/phase0"
+)
+
+// VersionedSignedValidatorRegistration contains a versioned SignedValidatorRegistrationV1.
+type VersionedSignedValidatorRegistration struct {
+	Version   spec.DataVersion
+	Bellatrix *apiv1.SignedValidatorRegistration
```

Comment:
This object isn't under the consensus specs versioning (this is stated explicitly in https://github.com/ethereum/builder-specs/blob/main/specs/builder.md#independently-versioned) so should use its own versioning (perhaps `spec.BuilderVersion`)